src/test_reference.jl

```diff
@@ -48,7 +48,8 @@ cases.
 ```
 """
 macro test_reference(reference, actual, kws...)
-    expr = :(test_reference(abspath(joinpath(Base.@__DIR__, $(esc(reference)))), $(esc(actual))))
+    dir = dirname(Base.source_path())
```

Comment:
I'm not sure what `Base.source_path()` does as it is not documented and not exported.
But I do know that there exists `Base.source_dir()`
which does basically  `dirname(Base.source_path())`

Reply:
But what is wrong with `@__DIR__`  anyway?
It hasn't been deprecated AFAIK.

Reply:
`@__DIR__` is evaluated in a context which is now not appropriate for the typical use cases. You're probably right about `source_dir`; I'll see if that works as well.

Reply:
A I see.
From NEWS.md

> @__FILE__ and @__DIR__ return information relative to the file that it was parsed from, rather than from the task-local SOURCE_PATH global when it was expanded.

Good catch.
This change will likely break one of my other packages.
I never noticed/gronked it before when I was reading NEWS.md